_rules/SC2-4-6-descriptive-headings.md

```diff
@@ -0,0 +1,170 @@
+<p>We are open Monday through Friday from 10 to 16</p>
+```
+
+#### Passed example 4
+
+Heading marked up with h-element that is a single character that describes the topic or purpose of the following section of content.
+
+```html
+<h1 class="target">A</h1>
+<dl>  
+<dt>airplane</dt>
+  <dd>a powered flying vehicle with fixed wings and a weight greater than that of the air it displaces.</dd>
+ <dt>apple</dt>
+  <dd>the round fruit of a tree of the rose family, which typically has thin green or red skin and crisp flesh.</dd>
+</dl>
+```
+
+#### Passed example 5
+
+Heading marked up with `role="heading"` that describes the topic or purpose of the following section of content. The heading is positioned off screen and is included in the Accessibility Tree.
+
+```html
+<span role="heading" style="position: absolute; top: -9999px; left: -9999px;">Opening Hours</span>
+<p style="position: absolute; top: -9999px; left: -9999px;">We are open Monday through Friday from 10 to 16</p>
+```
+
+#### Passed example 6
+
+Heading marked up with h-element that describes the topic or purpose of the following section of content. The heading is visible, but is not included in the Accessibility Tree.
+
+```html
+<h1 class="target" aria-hidden="true">Opening Hours</h1>
+<p>We are open Monday through Friday from 10 to 16</p>
+```
+
+### Failed
+
+#### Failed example 1
+
+Heading marked up with h-element that does not describe the topic or purpose of the following section of content.
+
+```html
+<h1 class="target">Weather</h1>
+<p>We are open Monday through Friday from 10 to 16</p>
+```
+
+#### Failed example 2
+
+Heading marked up with `role="heading"` that does not describe the topic or purpose of the following section of content.
+
+```html
+<span role="heading">Weather</span>
+<p>We are open Monday through Friday from 10 to 16</p>
+```
+
+#### Failed example 3
+
+Heading marked up with `role="heading"` that does not describe the topic or purpose of the following section of content. The heading is positioned off screen and is included in the Accessibility Tree.
+
+```html
+<span role="heading" style="position: absolute; top: -9999px; left: -9999px;">Weather</span>
+<p style="position: absolute; top: -9999px; left: -9999px;">We are open Monday through Friday from 10 to 16</p>
+```
+
+#### Failed example 4
+
+Heading marked up with h-element that dose not describe the topic or purpose of the following section of content. The heading is visible, but is not included in the Accessibility Tree.
+
+```html
+<h1 class="target" aria-hidden="true">Weather</h1>
+<p>We are open Monday through Friday from 10 to 16</p>
+```
+
+### Inapplicable
+
+#### Inapplicable example 1
+
+No heading.
+
+```html
+<p>We are open Monday through Friday from 10 to 16</p>
+```
+
+#### Inapplicable example 2
+
+Heading that is neither visible to users, nor included in the accessibility tree.
+
+```html
```

Comment:
one of the more common wrong situations is the empty headings:
```html
<h1></h1>
```
or
```html
<p role="heading" level="1"></p>
```

Reply:
Good point, added :)